apis/python/src/tiledbsoma/_dask/load.py

```diff
@@ -0,0 +1,184 @@
+# Copyright (c) TileDB, Inc. and The Chan Zuckerberg Initiative Foundation
+#
+# Licensed under the MIT License.
+
+"""Utilities for loading a TileDB-SOMA ``X`` matrix as a Dask Array.
+"""
+
+from __future__ import annotations
+
+from typing import (
+    TYPE_CHECKING,
+    Any,
+)
+
+import numpy as np
+import numpy.typing as npt
+import pyarrow as pa
+import scipy.sparse as sp
+from somacore.options import PlatformConfig, ResultOrder, ResultOrderStr, SparseNDCoords
+
+from tiledbsoma._dask.util import (
+    ChunkSize,
+    DaskConfig,
+    JoinIDs,
+    chunk_ids_sizes,
+    coords_to_joinids,
+    make_context,
+)
+from tiledbsoma._dense_nd_array import DenseNDArray
+from tiledbsoma._fastercsx import CompressedMatrix, Format
+from tiledbsoma._indexer import IntIndexer
+from tiledbsoma._sparse_nd_array import SparseNDArray
+from tiledbsoma.options._soma_tiledb_context import ConfigDict, ConfigVal
+
+if TYPE_CHECKING:
+    try:
+        import dask.array as da
+    except ImportError:
+        pass
+
+
+__all__ = [
+    # Re-export `DaskConfig` here, to save most users from needing separate `tiledbsoma._dask` import lines. It also
+    # appears in the public API of `SparseNDArrayRead.dask_array`, which imports it directly from
+    # `tiledbsoma._dask.util` (to avoid a circular dependency, since `SparseNDArray` is also referenced in this file).
+    "DaskConfig",
+    "load_daskarray",
+    "sparse_chunk",
+]
+
+
+def sparse_chunk(
+    block: npt.NDArray[tuple[JoinIDs, JoinIDs]],
+    block_info: dict[int | None, Any],
+    uri: str,
+    tiledb_config: ConfigDict,
+    tdb_concurrency: int | None = None,
+    format: Format = "csr",
+    result_order: ResultOrderStr = ResultOrder.AUTO,
+    platform_config: PlatformConfig | None = None,
+) -> sp.csr_matrix | sp.csc_matrix:
+    """Load a slice of a TileDB-SOMA ``X`` matrix, as a block of a CSR-backed Dask Array."""
+    shape = block_info[None]["chunk-shape"]
+    assert block.shape == (1, 1)
+    joinids = block[0, 0]
+    obs_joinids, var_joinids = joinids
+    obs_joinids = obs_joinids.astype("int64")
+    var_joinids = var_joinids.astype("int64")
+    # TODO: create sequence of tables for CM.from_soma; use record batches (idiomatic Arrow)
```

Comment:
As a general rule, a `TODO` comment should be accompanied by a link to a tracking task/story/issue